src/Arriba/Arriba.Composition/Composition/Host.cs

```diff
@@ -55,6 +55,14 @@ public static void AddArribaServices(this IServiceCollection services, ISecurity
             services.AddSingleton<ApplicationServer, ComposedApplicationServer>();
         }
 
+        private static void AddSecureDatabase(this IServiceCollection services)
+        {
+            var sp = services.BuildServiceProvider();
+            var databaseFactory = sp.GetService(typeof(DatabaseFactory)) as DatabaseFactory;
+
+            services.AddSingleton(databaseFactory.GetDatabase());
```

Comment:
Why do we need both a factory and a singleton instance registered?  Whatever is getting the Database injected could use the factory?

Reply:
We were registering both types, I kept doing the same but changed the source...

Reply:
I tracked the usage of DatabaseSecure and was easy to remove it's dependencies, thanks ;)